### PR TITLE
fix(dpg): path/url parameters not ordered by optional value

### LINKED ISSUE #2766: AutoRest C# generator generating DPG code with compilation errors
We are trying to generate DPG code for the swaggers listed [here](https://github.com/Azure/azure-sdk-for-net/blob/2ed2c1d5aa3576f556216a9da95600fa8b52a71f/sdk/devcenter/Azure.Developer.DevCenter/src/autorest.md) and we are getting [these](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1914087&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=a880e989-7d1a-5c96-a41f-d540b383cc43&l=69) errors for most of the methods:

> Error CS1737: Optional parameters must appear after all required parameters

For more details please see the failures in https://github.com/Azure/azure-sdk-for-net/pull/31636 PR

### PR DESCRIPTION
**Pull request by archerzz**
- revert previous commit to fix this issue
- sort path/url parameters, put required ones at the beginning, and put optional ones after body parameter
- add test cases

fix #2766

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs [hunk 1]
```diff
@@ -266,10 +266,6 @@ private void AddUriOrPathParameters(string uriPart, IReadOnlyDictionary<string,
                 {
                     AddParameter(text, requestParameter);
                 }
-                else
```
Comment (archerzz) [inline src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs:269]: In swagger validator, the parameters defined in URL path are checked against the `parameters` list. So here we can remove the exception safely: https://github.com/Azure/oav/blob/develop/documentation/oav-semantic-errors-reference.md#missing_path_parameter_definition

Comment (AlexanderSher) [reply src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs:269]: Do we have similar validation in CADL?

Comment (archerzz) [reply src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs:269]: > Do we have similar validation in CADL?

@AlexanderSher CADL compiler will check the path parameter.
image

File: test/TestProjects/Parameters-LowLevel/Parameters.json [hunk 2]
```diff
@@ -129,6 +129,77 @@
           }
         }
       }
+    },
+    "/test3/{id}/{name}": {
```
[the remark this answers is not in the document]
Comment (archerzz) [reply test/TestProjects/Parameters-LowLevel/Parameters.json:133]: @AlexanderSher Test case is added. Thanks.

**Review by AlexanderSher — APPROVED** [review]

**Pull request merged**